### ISSUE TITLE
ARM64 builds: templated_test_cases.cpp test fails

### ISSUE THREAD

**Issue by OdyX** [onqtam/doctest#119]
## Description
Debian's reproducibility test builds indicate that the arm64 build FTBFS:

https://tests.reproducible-builds.org/debian/rb-pkg/unstable/arm64/doctest.html

```
10/29 Test #10: templated_test_cases.cpp .........***Failed    0.08 sec
Files "/build/1st/doctest-1.2.7+repack0/examples/all_features/test_output/templated_test_cases.cpp.txt" to "/build/1st/doctest-1.2.7+repack0/obj-aarch64-linux-gnu/examples/all_features/CMakeFiles/temp_test_output.txt" are different.
==========================================================================
== CONTENTS OF /build/1st/doctest-1.2.7+repack0/examples/all_features/test_output/templated_test_cases.cpp.txt
==========================================================================
[doctest] run with "--help" for options
===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  vector stuff<std::vector<int>>

templated_test_cases.cpp(0) ERROR!
  CHECK( vec.size() == 20 )
with expansion:
  CHECK( 10 == 20 )

===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  multiple types<>

templated_test_cases.cpp(0) ERROR!
  CHECK( t2 != T2() )
with expansion:
  CHECK( 0 != 0 )

===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  multiple types<>

templated_test_cases.cpp(0) ERROR!
  CHECK( t2 != T2() )
with expansion:
  CHECK( 0 != 0 )

===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  multiple types<>

templated_test_cases.cpp(0) ERROR!
  CHECK( t2 != T2() )
with expansion:
  CHECK( 0 != 0 )

===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  bad stringification of type pair<int_pair>

templated_test_cases.cpp(0) ERROR!
  CHECK( t2 != T2() )
with expansion:
  CHECK( 0 != 0 )

===============================================================================
[doctest] test cases:     13 |      8 passed |      5 failed | 
[doctest] assertions:     17 |     12 passed |      5 failed |
[doctest] Status: FAILURE!
Program code.

==========================================================================
== CONTENTS OF /build/1st/doctest-1.2.7+repack0/obj-aarch64-linux-gnu/examples/all_features/CMakeFiles/temp_test_output.txt
==========================================================================
[doctest] run with "--help" for options
===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  signed integers stuff<char>

templated_test_cases.cpp(0) ERROR!
  CHECK( var == -1 )
with expansion:
  CHECK( 255 == -1 )

===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  vector stuff<std::vector<int>>

templated_test_cases.cpp(0) ERROR!
  CHECK( vec.size() == 20 )
with expansion:
  CHECK( 10 == 20 )

===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  multiple types<>

templated_test_cases.cpp(0) ERROR!
  CHECK( t2 != T2() )
with expansion:
  CHECK( 0 != 0 )

===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  multiple types<>

templated_test_cases.cpp(0) ERROR!
  CHECK( t2 != T2() )
with expansion:
  CHECK( 0 != 0 )

===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  multiple types<>

templated_test_cases.cpp(0) ERROR!
  CHECK( t2 != T2() )
with expansion:
  CHECK( 0 != 0 )

===============================================================================
templated_test_cases.cpp(0)
TEST CASE:  bad stringification of type pair<int_pair>

templated_test_cases.cpp(0) ERROR!
  CHECK( t2 != T2() )
with expansion:
  CHECK( 0 != 0 )

===============================================================================
[doctest] test cases:     13 |      7 passed |      6 failed | 
[doctest] assertions:     17 |     11 passed |      6 failed |
[doctest] Status: FAILURE!
Program code.

==========================================================================
== CONTENTS END
==========================================================================
CMake Error at /build/1st/doctest-1.2.7+repack0/scripts/cmake/exec_test.cmake:57 (message):
  Running
  '/build/1st/doctest-1.2.7+repack0/obj-aarch64-linux-gnu/examples/all_features/all_features
  --no-skipped-summary --no-version -sf=*templated_test_cases.cpp
  --dt-no-path-filenames=1 --dt-no-line-numbers=1 --dt-no-exitcode=1' ended
  with code 'Output is different from reference file!'

```

* doctest version: **v1.2.7**
* Operating System: **Debian sid**
* Compiler+version: **GNU 7.3.0**

**Comment by onqtam** [comment]
Weird - it seems that on that platform/compiler ```char``` is unsigned by default?!

this is the test case that fails:
https://github.com/onqtam/doctest/blob/master/examples/all_features/templated_test_cases.cpp#L15

That test case is instantiated with ```char```/```short```/```int``` and for each type I default construct a variable and substract 1 from that and I expect to get ```-1``` - but for ```char``` we get ```255``` ...

If you have any ideas (like what might be the cause for that - platform differences, etc.) I'm all ears.

**Comment by onqtam** [comment]
Actually char is not defined as signed/unsigned so I should change it to ```signed char```. Will do in a few days

**Comment by onqtam** [comment]
should be fixed in version 1.2.8
